archivistdht/private/eth/p2p/discoveryv5/protocol.nim

```diff
@@ -542,6 +508,78 @@ proc waitNodes(d: Protocol, fromNode: Node, reqId: RequestId):
     dht_message_requests_outgoing.inc(labelValues = ["no_response"])
     return err("Nodes message not received in time")
 
+proc pickUpUnknownNode(d: Protocol, fromId: NodeId, fromAddr: Address) {.async.} =
```

Comment:
You can use `{.async: (raises: []).}` to ensure the `asyncSpawn` below doesn't crash the process.